src/components/Tasks/TaskList.tsx

```diff
@@ -22,10 +20,70 @@ import Toolbar from './Toolbar';
 
 import { StoreState } from '../../store';
 
-const sortSelector = createSelector(
-  (entries: TaskType[]) => entries,
-  (entries) => entries.sort((a, b) => a.title.localeCompare(b.title))
-);
+function sortCompleted(a: TaskType, b: TaskType) {
+  return (!!a.finished === !!b.finished) ? 0 : (a.finished) ? 1 : -1;
+}
+
+function sortLastModifiedDate(aIn: TaskType, bIn: TaskType) {
+  const a = aIn.lastModified?.toJSDate() ?? new Date(0);
+  const b = bIn.lastModified?.toJSDate() ?? new Date(0);
+  return (a > b) ? -1 : (a < b) ? 1 : 0;
+}
+
+function sortDueDate(aIn: TaskType, bIn: TaskType) {
+  const a = aIn.dueDate?.toJSDate() ?? new Date(0);
+  const b = bIn.dueDate?.toJSDate() ?? new Date(0);
+  return (a > b) ? -1 : (a < b) ? 1 : 0;
+}
+
+function sortPriority(aIn: TaskType, bIn: TaskType) {
+  const a = aIn.priority || Infinity;
```

Comment:
Are you intentionally turning zero here to infinity?

Reply:
Yes. The way priorities work is kind of dumb. 1 is the highest priority and 9 is the lowest, but 0 is undefined. The comparator math doesn't work out unless you make undefined the highest amount (else it sorts undefined priorities to the top).

Reply:
Maybe add a comment before this line then? I just feel like it's one of those things that someone down the line will see and think it was a mistake and change it to `??` (which is wrong).